internal/codeintel/autoindexing/service.go

```diff
@@ -470,6 +484,18 @@ func (s Service) queueIndexForRepositoryAndCommit(ctx context.Context, repositor
 		return nil, nil
 	}
 
+	if !force {
+		for _, index := range indexes {
+			isQueued, err := s.store.IsQueuedRootIndexer(ctx, repositoryID, commit, index.Root, index.Indexer)
+			if err != nil {
+				return nil, errors.Wrap(err, "dbstore.IsQueuedRootIndexer")
+			}
+			if isQueued {
+				return nil, nil
+			}
+		}
+	}
```

Comment:
```suggestion
		filtered := indexes[:0]
		for _, index := range indexes {
			isQueued, err := s.store.IsQueuedRootIndexer(ctx, repositoryID, commit, index.Root, index.Indexer)
			if err != nil {
				return nil, errors.Wrap(err, "dbstore.IsQueuedRootIndexer")
			}
			if !isQueued {
				filtered = append(filtered, index)
			}
		}
		indexes = filtered
```

Do we want to filter here instead of throwing out the entire batch?